nep-oracles.mediawiki

```diff
@@ -0,0 +1,54 @@
+<pre>
+  NEP: <to be assigned>
+  Title: Neo oracles
+  Author: Shargon and Belane
+  Type: Standard
+  Status: Draft
+  Created: 2018-11-22
+</pre>
+
+==Abstract==
+
+The objective of this proposal is to be the first chain capable of not needing oracles to interact with non-deterministic content. Our mission is to make NEO capable of performing REST API calls and other types of non-deterministic functionalities that can be extended in the future.
+
+==Motivation==
+
+Nowadays it is necessary to have access to information sources external to the Smart Contract for the performance of certain tasks (non-deterministic tasks). That is why it is necessary to give the option to run non-deterministic content safely and without the need of a centralized oracle.
+
+==Specification==
+
+Information will be stored in each block header. This information will be necessary, both for the correct execution of the transactions contained in it, and general information for positioning the block within the blockchain.
+
+[[File:oracles/oracles-image-1.png|center|header]]
+
+A section called "MetaData" will be added to the block that will store, following the key-value structure, the information necessary for the execution that will allow the persistence of the result over time. ''This header (or the entire block) can be compressed using Gzip.''
+
+In order to provide the system with a non-deterministic execution, Smart Contracts will be provided with a set of Access Proxies to all those elements that could cause an unstable result (through SYSCALL).
+
+The result of the execution of a non-deterministic function will be stored in the metadata of the block header, allowing the future persistence of the result thus giving the system determinism in time. The execution of a Smart Contract will check if the result has been stored, in which case, the value will be reused. Otherwise, the complete execution of the contract will be performed and the result will be stored in the metadata of the block according to the following cycle:
+
+[[File:oracles/oracles-image-2.png|center|flow]]
+
+Another option is to include the information of the results of the Oracles in the remark field of the transaction itself, being able to validate each TX for specific results.

```

Comment:
As I understood until, I think I prefer in the TX, but I need to understand it further.